src/TestInfrastructure/Helpers/MefTestHelpers.cs

```diff
@@ -54,6 +54,44 @@ public static Export CreateExport<T>(object exportInstance, string contractName
             return export;
         }
 
+        /// <summary>
+        /// Overload that imports using a SingleObjectImporter and checks that the expected
+        /// implementing type was imported, and that the expected CreationPolicy has been applied
+        /// </summary>
+        public static SingleObjectImporter<TImportType> CheckTypeCanBeImported<TTypeToCheck, TImportType>(
+            CreationPolicy expectedCreationPolicy,
+            params Export[] requiredExports)
+            where TImportType : class
+            where TTypeToCheck : class
+        {
+            if (expectedCreationPolicy == CreationPolicy.Any)
+            {
+                throw new ArgumentOutOfRangeException("Test setup error: CreationPolicy.Any is not supported (we don't use it anywhere currently)");
+            }
+
+            // First, just try to import the type
+            var importer1 = new SingleObjectImporter<TImportType>();
+            Compose(importer1, typeof(TTypeToCheck), requiredExports);
+            importer1.AssertImportIsNotNull();
+            importer1.AssertImportIsInstanceOf<TTypeToCheck>();
+
+            // Now import the type again, doing the full check that the requiredExports
+            // are actually required
+            var importer2 = CheckTypeCanBeImported<TTypeToCheck, TImportType>(requiredExports);
+
+            // Check the expected creation policy was used
```

Comment:
Checking the creation policy is simple - import the type twice, and see if we get the same instance or not.

Reply:
Actually, Why do you not reuse the code we already have in this file? CheckIsNonSharedMefComponent() and CheckIsSingletonMefComponent() ?

Reply:
Good point. I remembered you'd added those recently, then forgot to go and look at them.
I'll delete my new test method since it is more complicated and use your existing helpers.